2019/5xxx/CVE-2019-5624.json

```diff
@@ -1,18 +1,127 @@
-            }
+  "data_type": "CVE",
+  "data_format": "MITRE",
+  "data_version": "4.0",
+  "generator": {
+    "engine": "Vulnogram 0.0.6"
+  },
+  "CVE_data_meta": {
+    "ID": "CVE-2019-5624",
+    "ASSIGNER": "cve@rapid7.com",
+    "DATE_PUBLIC": "2019-04-24T18:00:00.000Z",
+    "TITLE": "Rapid7 Metasploit Framework Zip Import Directory Traversal",
+    "AKA": "",
+    "STATE": "PUBLIC"
+  },
+  "source": {
+    "defect": [
+      "MS-3923"
+    ],
+    "advisory": "",
+    "discovery": "UNKNOWN"
+  },
+  "affects": {
+    "vendor": {
+      "vendor_data": [
+        {
+          "vendor_name": "Rapid7",
+          "product": {
+            "product_data": [
+              {
+                "product_name": "Metasploit Framework",
+                "version": {
+                  "version_data": [
+                    {
+                      "version_name": "4.14.0",
+                      "version_affected": "<=",
+                      "version_value": "4.14.0",
+                      "platform": ""
+                    }
+                  ]
+                }
+              }
+            ]
+          }
+        }
+      ]
+    }
+  },
+  "problemtype": {
+    "problemtype_data": [
+      {
+        "description": [
+          {
+            "lang": "eng",
+            "value": "CWE-22"
+          },
+          {
+            "lang": "eng",
+            "value": "Improper Limitation of a Pathname to a Restricted Directory ('Path Traversal')"
+          }
         ]
+      }
+    ]
+  },
+  "description": {
+    "description_data": [
+      {
+        "lang": "eng",
+        "value": "Rapid7 Metasploit Framework suffers from an instance of CWE-22, Improper Limitation of a Pathname to a Restricted Directory ('Path Traversal') in the Zip import function of Metasploit. Exploiting this vulnerability can allow an attacker to execute arbitrary code in Metasploit at the privilege level of the user running Metasploit."
+      },
+      {
+        "lang": "eng",
+        "value": "This issue affects:\nRapid7 Metasploit Framework version 4.14.0 and prior versions."
+      }
+    ]
+  },
+  "references": {
+    "reference_data": [
+      {
+        "refsource": "CONFIRM",
+        "url": "https://github.com/rapid7/metasploit-framework/pull/11716",
+        "name": "https://github.com/rapid7/metasploit-framework/pull/11716"
+      },
+      {
+        "refsource": "CONFIRM",
+        "url": "https://help.rapid7.com/metasploit/release-notes/archive/2019/04/#20190416",
+        "name": "https://help.rapid7.com/metasploit/release-notes/archive/2019/04/#20190416"
+      },
+      {
+        "refsource": "MISC",
+        "url": "https://blog.doyensec.com/2019/04/24/rubyzip-bug.html",
+        "name": "https://blog.doyensec.com/2019/04/24/rubyzip-bug.html"
+      }
+    ]
+  },
+  "configuration": [],
+  "impact": {
+    "cvss": {
+      "version": "3.0",
+      "attackVector": "LOCAL",
```

Comment:
Local due to needing to get the special zip file onto the box before import, correct? That step aside I believe you can call these import functions via API so wouldn't need to be local. But that first step makes sense to increase this requirement.

Reply:
The zip has to end up on the local filesystem somehow, yep, so it feels awfully local to me. The likely attack is to get your Metasploit-using friend to import your evil zip file; if you already have API access, you already have a bunch of privileges so don't need to do all this.